tests/components/home_plus_control/test_init.py

```diff
@@ -0,0 +1,962 @@
+    """
+
+    # Register the mock responses
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL,
+        text=plant_data,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210" + PLANT_TOPOLOGY_RESOURCE,
+        text=plant_topology,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210",
+        text=plant_modules,
+    )
+
+    # Load the entry
+    mock_config_entry.add_to_hass(hass)
+    assert await setup.async_setup_component(
+        hass,
+        "home_plus_control",
+        {
+            "home_plus_control": {
+                CONF_CLIENT_ID: CLIENT_ID,
+                CONF_CLIENT_SECRET: CLIENT_SECRET,
+                CONF_SUBSCRIPTION_KEY: SUBSCRIPTION_KEY,
+            },
+        },
+    )
+    await hass.async_block_till_done()
+
+    # The setup of the integration calls the API 3 times
+    assert aioclient_mock.call_count == 3
+
+    # The component has been loaded
+    assert isinstance(
+        hass.data[DOMAIN]["home_plus_control_entry_id"][API],
+        api.HomePlusControlAsyncApi,
+    )
+    assert mock_config_entry.state == config_entries.ENTRY_STATE_LOADED
+
+    # Check the entities and devices - all entities should be there
+    await entity_assertions(
+        hass,
+        num_exp_entities=5,
+        expected_entities={
+            "switch.dining_room_wall_outlet": True,
+            "switch.kitchen_wall_outlet": True,
+        },
+    )
+    for test_entity_uid in hass.data[DOMAIN]["home_plus_control_entry_id"][ENTITY_UIDS]:
+        await one_entity_assertion(hass, test_entity_uid, True)
+
+    # Attempt to update the data, but plant topology update fails
+
+    # Reset the mock responses
+    aioclient_mock.clear_requests()
+    # Register the mock responses
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL,
+        text=plant_data,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210" + PLANT_TOPOLOGY_RESOURCE,
+        text=plant_topology_reduced,
+        exc=asyncio.TimeoutError,
+    )
+    aioclient_mock.get(
+        PLANT_TOPOLOGY_BASE_URL + "123456789009876543210",
+        text=plant_modules_reduced,
+    )
+
+    with patch(
+        "homeassistant.components.home_plus_control.api.HomePlusControlAsyncApi._should_check",
+        return_value=True,
+    ) as mock_check:
+        async_fire_time_changed(
+            hass, dt.datetime.now(dt.timezone.utc) + dt.timedelta(seconds=300)
+        )
+        await hass.async_block_till_done()
+        assert len(mock_check.mock_calls) == 3
+
+    # Check for plant, topology and module status
+    await entity_assertions(
+        hass,
+        num_exp_entities=5,
+        expected_entities={
+            "switch.dining_room_wall_outlet": True,
+            "switch.kitchen_wall_outlet": True,
+        },
+    )
+
+    # This entity has not returned a status, so appears as unavailable
```

Comment:
The entity state should be tested in a module for the platform. Please move those tests to a module `test_switch.py`.

Reply:
OK, I have moved these tests to the `test_switch.py` module and I am working on cleaning these up.